### ISSUE TITLE
Is Self.GET of a `/partial/` URI needed for blending?

### ISSUE THREAD

**Issue by warpech** [Starcounter/Starcounter#4118]
Today I had a Hangouts call with @jwillmer. 

One of the surprises for him was that it is needed to define blending points explicitly using a combination of `Self.GET` and `Handle.GET(".../partials/..."`.

Below excerpt from Producs app shows this:

```cs
Handle.GET("/products/product/{?}", (string objectId) => {
    StandalonePage master = (StandalonePage)Self.GET("/Products/standalone");
    master.CurrentPage = Db.Scope<Json>(() => Self.GET<EditProductPage>("/Products/partials/product/" + objectId));
    return master;
});

Handle.GET("/products/partials/product/{?}", (string objectId) => {
    return Db.Scope(() => {
        var product = (Product) DbHelper.FromID(DbHelper.Base64DecodeObjectID(objectId));
        var page = new EditProductPage() {Data = product};

        /* ... */

        return page;
    });
});
```

It is not obvious why such obscure combination of two handlers is needed.

Is there a way we could support implicit blending points every time you nest `Json` in `Json`? Potentially this could work:

```cs
Handle.GET("/products/product/{?}", (string objectId) => {
    StandalonePage master = (StandalonePage)Self.GET("/Products/standalone");
    master.CurrentPage = Db.Scope(() => new EditProductPage(objectId));
    return master;
});

[Blender.Uri("/Products/partials/product/{?}")]
partial class EditProductPage : Json, IBound<Product>
{
/* ... */
```

cc @chrhol

**Comment by warpech** [comment]
cc @Starcounter-Jack, @miyconst, @alemoi

**Comment by alemoi** [comment]
I suppose the blending can be declarative into JSON somehow. I have a question: should it be possible that the partial handler is called directly from browser and blend with other handlers? We discussed on the meeting.

**Comment by jwillmer** [comment]
You can use `SelfOnly` as `HandlerOption` to prevent this ([doc](https://docs.starcounter.io/guides/network/handling-http-requests/)) :point_up: :laughing:

**Comment by chrhol** [comment]
> It is not obvious why such obscure combination of two handlers is needed.

Isn't this what we discussed (at length) previous week? There we said that the way to enable blending (blending points) is to retrieve an instance of a (partial) viewmodel  by doing a request (any request, not only `Self.GET`). 

We choose this pattern, instead of just creating the viewmodel or calling a method directly, to enable support for blending when we can have multiple codehosts which runs apps that should be blended.

> Is there a way we could support implicit blending points every time you nest Json in Json? Potentially this could work:

I don't really like this. Then we're back at supporting a pattern that would make it more difficult to explain and document and that hides the statement above. Not to say that it would mean that `Starcounter.XSON` have to be aware of request/response handling (doing the `Self.GET` under the hood). 

So if the question is, could we add this? Maybe. *Should* we add this in Starcounter and support in our samples, then I think no, we shouldn't. But we should make sure that the appdeveloper/s could add this functionality themselves in their code, if they want to have this pattern.

But then again, this is my personal opinion 😄 I think blending is kind of a complex topic, when it comes down to actual code (and not a high-level explanation), and make some code in apps not obvious to follow.  But I think trying to hide the fact that you need to do a request is the wrong way to go.

**Comment by chrhol** [comment]
> I have a question: should it be possible that the partial handler is called directly from browser and blend with other handlers? 

Yes, if I remember correctly, I asked this question as well last week in the meeting. And the conclusion was that *any* request, external and internal, could be a possible blending point.
